docs: retroactive MCP release notes for Sync 2.1.7 and Portal 1.17.1 (#1951)
TykTechnologies/tyk-docs: developer-support/release-notes/sync.mdx

#### Changelog
<a id="Changelog-v2.1.7" data-scroll-offset></a>

##### Added

<AccordionGroup>

> <Accordion title='MCP Proxy support in tyk-sync'>

Reviewer: ```suggestion
<Accordion title='MCP Proxy support'>
```

Can you please stop including the name of the component in the release notes? It's unnecessary - and when it is required, please use the product name, not the git repository name.

> Tyk Sync now recognises MCP proxies as a distinct resource type alongside OAS APIs. The `dump` and `sync` commands handle MCP proxies correctly:

Reviewer: Never "OAS APIs"... how many times? :-(

On this occasion, I'm not sure why we are talking about "MCP vs OAS", given that Sync also supports Tyk Classic. It's just unnecessary noise.

> Tyk Sync now recognises MCP proxies as a distinct resource type alongside OAS APIs. The `dump` and `sync` commands handle MCP proxies correctly:

Reviewer: ```suggestion
Tyk Sync recognises [MCP proxies](/ai-management/mcp-gateway/managing-mcp-proxies) as a distinct resource type.

The `dump` and `sync` commands have been extended to handle MCP proxies:
```

> - Dump writes MCP proxies to `mcp-{id}.json` files, separate from `oas-{id}.json` files for OAS APIs.

Reviewer: ```suggestion
- Dump writes MCP proxies to `mcp-{id}.json` files.
```

> - Command output reports MCP proxies and OAS APIs as separate counts: `X OAS APIs` and `Y MCP Proxies`.

Reviewer: ```suggestion
- Command output reports a separate count for MCP proxies.
```

> - Create, update, and delete operations are routed to the `/api/mcps` endpoint rather than `/api/apis/oas`.

Reviewer: ```suggestion
- Create, update, and delete operations are routed to the dedicated `/api/mcps` endpoint.
```

- MCP proxies are included automatically when fetching all resources.

> No changes to your existing workflow are required — MCP proxies are handled alongside your OAS APIs in the same `tyk-sync` commands.

Reviewer: ```suggestion
No changes to your existing workflow are required — MCP proxies are handled using the same `tyk-sync` commands.
```

<AccordionGroup>

> <Accordion title='MCP proxies being recreated as OAS APIs after sync'>

Reviewer: This seems to be relevant, as MCP support is only being added in this release? Seems to be a bug that came up during development.

#### Changelog
<a id="Changelog-v2.1.7" data-scroll-offset></a>

> ##### Added

Reviewer 2: @andyo-tyk @challiscj 
Is this for 2.1.7 or 2.1.8?

Reviewer: AFAIK it was introduced in 2.1.7 -> but there was an error with not aligning the API definition, so we had to go straight to 2.1.8.

Reviewer 2: Ok, I'll keep this here then.

<AccordionGroup>

<Accordion title='MCP Proxy support'>

> Tyk Sync recognises [MCP proxies](/ai-management/mcp-gateway/managing-mcp-proxies) as a distinct resource type.

Reviewer 2: ```suggestion
Tyk Sync recognises [MCP proxies](/ai-management/mcp-gateway/managing-proxies) as a distinct resource type.
```
